
enum Command{
   GET_CHARM_REQ = 1001;
   GET_CHARM_RESP = 1002; 
   INTEREST_REQ = 1003;
   INTEREST_RESP = 1004;  
};

enum Status{
   SUCCESS = 0;
   DB_FAIL = 101;
};

message Person{
   required string phone = 1;
   optional string name = 2;
   optional string message = 3;
   optional string sendmsg = 4;
}

message GetCharmRequest{
   required string phone = 1;
   optional string name = 2;
}

message GetCharmRespone{
   required int32 charm_mark = 1;
   repeated Person interest_person = 2;
}

message InterestRequest{
   required string phone = 1;
   required string to_phone = 2;
   optional string name = 3;
   optional string message = 4;   
}

message InterestRespone{
 
}

message StringPair{
   required string key = 1;
   required string value = 2;
}

message GetRecommendRequest{
   required string phone = 1;
   repeated StringPair fields= 2;
}

message GetRecommendResponse{
   repeated StringPair fields= 1;
}

message Request{
   optional GetCharmRequest get_charm_req = 1;
   optional InterestRequest interest_req =2;
   optional GetRecommendRequest get_recommand_req = 3;
}

message Response{
   required int32 status = 1;
   optional GetCharmRespone get_charm_resp = 2;
   optional InterestRequest interest_resp = 3;
   optional GetRecommendResponse get_recommend_resp = 4;
}
